spec/controllers/grade_entry_forms_controller_spec.rb

```diff
@@ -0,0 +1,108 @@
+require 'spec_helper'
+
+describe GradeEntryFormsController do
+
+  before :each do
+    # Authenticate user is not timed out, and has administrator rights.
+    allow(controller).to receive(:session_expired?).and_return(false)
+    allow(controller).to receive(:logged_in?).and_return(true)
+    allow(controller).to receive(:current_user).and_return(build(:admin))
+
+  end
+
+  let(:grade_entry_form) { create(:grade_entry_form) }
+
+
+
+context 'CSV_Uploads' do
+
+    before :each do
+      @file_without_extension = fixture_file_upload('spec/fixtures/files/grade_entry_upload_empty_file', 'text/xml')
+      @file_wrong_format = fixture_file_upload('spec/fixtures/files/grade_entry_upload_wrong_format.xls', 'text/xls')
+      @file_bad_csv = fixture_file_upload('spec/fixtures/files/grade_entry_upload_bad_csv.csv', 'text/xls')
+      @file_wrong_columns = fixture_file_upload('spec/fixtures/files/grade_entry_upload_wrong_columns.csv', 'text/csv')
+    end
+    
+    
+    it 'does not accept a csv file with wrong data columns' do
+      post :csv_upload, id: grade_entry_form, :upload => { :grades_file => @file_wrong_columns}
+      expect(response.status).to eq(302)
+      expect(flash[:error]).to_not be_empty
+      expect(response).to redirect_to(grades_grade_entry_form_path(grade_entry_form, locale: 'en'))
+    end
+    
+    it 'does not accept a file with no extension' do
+      post :csv_upload, id: grade_entry_form, :upload => { :grades_file => @file_without_extension}
```

Comment:
Line is too long. [99/80]<br>Use the new Ruby 1.9 hash syntax.<br>Space inside } missing.